src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/CompletionHandler.cs

```diff
@@ -123,6 +123,28 @@ internal class CompletionHandler : IRequestHandler<CompletionParams, SumType<Com
                 result = SetResolveData(result.Value, serverKind);
             }
 
+            // This is a bit fraught. Basically in the scenario That someone types "@<slight pause>{" they'll get @__builder because '{' is a commit character.
+            // Our solution is to make the results in this case empty, which is a bit of a lie.
+            // It's the best we can do for now, if a user complains we can revisit.
+            if (result.HasValue && request.Context.TriggerCharacter != null && request.Context.TriggerCharacter.Equals("@"))
```

Comment:
You're going to also need to update the CompletionList to indicate IsIncomplete = true to ensure we're re-queried for completion after typing `@`. Otherwise when the editor becomes more spec-abiding it wont requery us after typing `@`. Sounds like VS is re-querying completions on every keystroke currently